league_rpc_linux/processes/LCU_Thread.py

```diff
@@ -0,0 +1,203 @@
+
+    data["queue"] = lobbyQueueInfo["name"]
+    data["max_players"] = lobbyQueueInfo["maximumParticipantListSize"]
+    data["map_id"] = lobbyQueueInfo["mapId"]
+    data["gamemode"] = lobbyQueueInfo["gameMode"]
+    data["lobbyQueueIsRanked"] = lobbyQueueInfo["isRanked"]
+
+    await update_rpc()
+
+
+@connector.ws.register("/lol-ranked/v1/current-ranked-stats/", event_types=("UPDATE",))
+async def ranked(connection, event):
+    data["summoner_rank"] = (
+        event.data["highestRankedEntry"]["tier"]
+        + " "
+        + event.data["highestRankedEntry"]["division"]
+        + ": "
+        + str(event.data["highestRankedEntry"]["leaguePoints"])
+        + " LP"
+    )
+
+    await update_rpc()
+
+
+@connector.ws.register("/lol-matchmaking/v1/search", event_types=("UPDATE",))
+async def matchmaking(connection, event):
+    dataSearchState = event.data.get("searchState", None)
+
+    if data["queueState"] == dataSearchState:
+        # No change in state, only timer updated.
+        # print(f"Debug: {dataSearchState} == {data['queueState']}")
+        return
+    data["queueState"] = event.data.get("searchState", None)
+
+    print("Matchmaking has been updated.")
+    await update_rpc()
+
+
+async def gather_based_data(connection):
+    print("Gathering base data.")
+    summonerDataRaw = await connection.request(
+        "GET", "/lol-summoner/v1/current-summoner"
+    )
+    summonerData = await summonerDataRaw.json()
+    data["summoner_name"] = summonerData["displayName"]
+    data["level"] = summonerData["summonerLevel"]
+    data["summoner_id"] = summonerData["summonerId"]
+
+    rankedDataRaw = await connection.request(
+        "GET", "/lol-ranked/v1/current-ranked-stats/"
+    )
+    rankedData = await rankedDataRaw.json()
+    data["summoner_rank"] = (
+        rankedData["highestRankedEntry"]["tier"]
+        + " "
+        + rankedData["highestRankedEntry"]["division"]
+        + ": "
+        + str(rankedData["highestRankedEntry"]["leaguePoints"])
+        + " LP"
+    )
+
+    lobbyRawData = await connection.request(
+        "GET", "/lol-gameflow/v1/gameflow-metadata/player-status"
+    )
+    lobbyData = await lobbyRawData.json()
+    data["queue_id"] = lobbyData["currentLobbyStatus"]["queueId"]
+    data["lobby_id"] = lobbyData["currentLobbyStatus"]["lobbyId"]
+    data["players"] = len(lobbyData["currentLobbyStatus"]["memberSummonerIds"])
+
+    if data["queue_id"] == -1:
+        # Not in Lobby
+        return
+
+    lobbyQueueInfoRaw = await connection.request(
+        "GET", "/lol-game-queues/v1/queues/" + str(data["queue_id"])
+    )
+    lobbyQueueInfo = await lobbyQueueInfoRaw.json()
+    data["queue"] = lobbyQueueInfo["name"]
+    data["max_players"] = lobbyQueueInfo["maximumParticipantListSize"]
+    data["map_id"] = lobbyQueueInfo["mapId"]
+    data["gamemode"] = lobbyQueueInfo["gameMode"]
+    data["lobbyQueueIsRanked"] = lobbyQueueInfo["isRanked"]
+
+
+async def update_rpc():
```

Comment:
Why don't we update rpc, inside of the websocket functions?

Reply:
as in some cases different data is needed. - also this would sometimes override the rpc info with data we dont want overridden. 

Like being in champ select and the final "DELETE" from the queue is executed, it would remove the queue info and display the basic "idle in client" again, because it was the last "update" we got. - Until the next ChampSelect update of course. 

It also allows for "rate-limiting" RPC updates. - some events get executed mutliple times a second (Opening a lobby triggers 4 or 5 lobby updates, for which each of those would trigger rpc updates), we dont want to spam the RPC. 

In my current state, i've rate limited the rpc updates to 1 per second. - See delayUpdate()